packages/react-ui-components/src/components/SchemaForm/Widgets/CheckboxesWidget/utils.ts

```diff
@@ -0,0 +1,92 @@
+import { IRadioButton } from '@kelvininc/ui-components';
+import { ALL_BUTTON_VALUE } from './config';
+import { ICheckboxConfig } from './types';
+
+export const buildRadioButtons = <T extends { label: string; value: string }>(options: T[], enumDisabled: string[], { multiple, allButton }: ICheckboxConfig): IRadioButton[] =>
+	options.reduce<IRadioButton[]>(
+		(accumulator, { label, value }) => [
+			...accumulator,
+			{
+				label,
+				value,
+				disabled: enumDisabled.indexOf(value) !== -1
+			}
+		],
+		multiple && allButton
+			? [
+					{
+						label: 'All',
+						value: ALL_BUTTON_VALUE
+					}
+			  ]
+			: []
+	);
+
+export const buildSelectedRadioButtons = <T extends { label: string; value: string }>(
+	selectedOptions: string[],
+	allOptions: T[],
+	{ multiple, allButton }: ICheckboxConfig
+): { [key: string]: boolean } => {
+	if (multiple && allButton) {
+		// check all buttons are selected
+		if (selectedOptions.length === allOptions.length) {
+			return { [ALL_BUTTON_VALUE]: true };
+		}
+	}
+
+	return selectedOptions.reduce<{ [key: string]: boolean }>((accumulator, selectOption) => {
+		accumulator[selectOption] = true;
+
+		return accumulator;
+	}, {});
+};
+
+export const toggleSelectedOptions = <T extends { label: string; value: string }>(
+	selectedOptionValue: string,
+	selectedOptions: string[],
+	allOptions: T[],
+	{ multiple, allButton, minItems, maxItems }: ICheckboxConfig
+): string[] => {
+	if (!multiple) {
+		if (selectedOptions.includes(selectedOptionValue)) {
+			return [];
+		}
+
+		return [selectedOptionValue];
+	}
+
+	const allButtonsSelected = selectedOptions.length === allOptions.length;
+
+	// check if the user clicked on 'All' button
+	if (allButton) {
+		if (selectedOptionValue === ALL_BUTTON_VALUE) {
+			// reset to the first option selected
+			if (allButtonsSelected) {
+				return [allOptions[0].value];
```

Comment:
I think you should only force this selection if the field is required.

Reply:
This makes sense. I'll add a condition here to check `required || minItems > 0`.